File: 4-ruby/assignments/0102/wizard_ninja_samurai/human.rb
class Human
	attr_accessor :strength, :intelligence, :stealth, :health

	def initialize (strength: 3, intelligence: 3, stealth: 3, health: 100)
		@strength = strength
		@intelligence = intelligence
		@stealth = stealth
		@health = health
	end

	def attack (target)
		if target.class.ancestors.include?(Human)
			puts "Tis' but a scratch"
			target.health -= 10
			puts "Target health is now: #{target.health}"
		else
			puts "Not human ancestor"
		end
	end
end

File: 6-csharp/assignments/w01/d02/WizardNinjaSamurai/Human.cs
using System;

namespace WizardNinjaSamurai
{
    class Human
    {
        public string Name;
        public int Strength;
        public int Intelligence;
        public int Dexterity;
        protected int health;

        public int Health
        {
            get { return health; }
        }

        public Human(string name)
        {
            Name = name;
            Strength = 3;
            Intelligence = 3;
            Dexterity = 3;
            health = 100;
        }

        public Human(string name, int strength, int intelligence, int dexterity, int he)
        {
            Name = name;
            Strength = strength;
            Intelligence = intelligence;
            Dexterity = dexterity;
            health = he;
        }

        public int TakeDamage(int damage)
        {
            health -= damage;
            return health;
        }

        public virtual int Attack(Human target)
        {
            Console.WriteLine("Attack");
            int damage = 5 * Strength;
            TakeDamage(damage);
            Console.WriteLine($"Health of target is now {target.health}");
            return target.Health;
        }
    }
}